Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/feedback/FeedbackDialogFragment.kt

```diff
@@ -0,0 +1,100 @@
+package org.cornelldti.density.density.facilitydetail.feedback
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import androidx.viewpager2.widget.ViewPager2
+import org.cornelldti.density.density.R
+
+
+class FeedbackDialogFragment : DialogFragment() {
+
+    private lateinit var viewPager: ViewPager2
+    private val BROADCAST_ACTION = "BROADCAST_ACTION"
+    private var FIRST_INPUT = 0
+    private var SECOND_INPUT = 0
+    private var THIRD_INPUT = ""
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_feedback_dialog, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewPager = view.findViewById(R.id.viewPager)
+        viewPager.isUserInputEnabled = false
+        setUpViewPager()
+
+        dialog?.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+        dialog?.setCancelable(false)
+        dialog?.setCanceledOnTouchOutside(false)
+    }
+
+    private fun setUpViewPager(){
+        val adapter = FeedbackStateAdapter(childFragmentManager, lifecycle)
+
+        adapter.addFragment(FeedbackFirstFragment())
+        adapter.addFragment(FeedbackSecondFragment())
+        adapter.addFragment(FeedbackThirdFragment())
+        adapter.addFragment(FeedbackFourthFragment())
+
+        viewPager.adapter = adapter
+    }
+
+    var broadcastReceiver: BroadcastReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context?, intent: Intent) {
+            if (intent.action != null && intent.action == BROADCAST_ACTION) {
+                dialog!!.dismiss()
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        context?.let {
+            LocalBroadcastManager.getInstance(
+                    it).registerReceiver(broadcastReceiver, IntentFilter(BROADCAST_ACTION))
+        }
+    }
+
+    override fun onPause() {
+        super.onPause()
+        context?.let { LocalBroadcastManager.getInstance(it).unregisterReceiver(broadcastReceiver) }
+    }
+
+    fun setFirstInput(int : Int){
```

Comment:
Maybe these functions and associated variable names should be more descriptive if they hold the data for each survey question. Also I think you can use/customize Kotlin's getters and setters for properties instead of writing functions `getSomething` and `setSomething`

Reply:
good suggestion - probably not worth even making them `private` in the first place if they're meant to be mutable from other places